intents/aspects.py

```diff
@@ -0,0 +1,156 @@
+
+    Returns:
+        The updated table after aplying the condition, in pandas.dataframe type
+    """
+    if slices is None:
+    	return table
+    num_rows = table.shape[0]
+    for row in range(num_rows):
+        slice_match = True
+        for key in slices:
+            if table.loc[row, key] != slices[key]:
+                slice_match = False
+        if slice_match is not True:
+            table = table.drop([row])
+
+    table = table.reset_index()    # some indices get deleted after slicing
+    table = table.drop(['index'], axis=1)    # droping the new columnn named 'index' created after reset_index call
+    return table
+
+def corp_other_dimensions(table, dimensions, metric):
+    """This function removes the columns that are not in dimensions and are not equal to metric
+
+    Args:
+        table: Type-pandas.dataframe
+        dimensions: Type-list of str
+        metric: Type-string
+
+    Returns:
+        Returns the table as a dataframe obj after removing the rest of the columns.
+    """
+
+    current_columns = table.columns
+    for column in current_columns:
+        if (column != metric) and (column not in dimensions) :
+            table = table.drop([column], axis=1)
+
+    return table
+
+def count_distinct(values):
+    """ This function returns the number of distinct entries in values.
+    The values are inserted into a set and it's size is returned.
+
+    Args:
+        values: Type - anything that can be iterated over
+
+    Returns:
+        Count of distinct values.
+    """
+    distinct_set = set()
+    for value in values:
+        distinct_set.add(value)
+    return len(distinct_set)
+
+def group_by(table, dimensions, summary_operator):
+    """Groups the column by the columns in dimensions
+    Basically makes a map in which keys contain dimensions and values are evaluated after applying the summary operator.
+
+    Args:
+        table: Type-pandas.dataframe
+            It has the contents of the csv file
+        dimensions: Type-list
+            It contains the names of columns according to which we group
+        summary_operator: Type-str
+            It denotes the summary operator
+
+    Returns:
+        Returns the table as a dataframe obj after applying grouping
+    """
+    if summary_operator is None:
+    	return table
+    if summary_operator == 'sum':
+        table = table.groupby(dimensions).sum()
+    if summary_operator == 'average':
+        table = table.groupby(dimensions).mean()
+    if summary_operator == 'count':
+        table = table.groupby(dimensions).count()
+    if summary_operator == 'max':
+        table = table.groupby(dimensions).describe()
+    if summary_operator == 'min':
+        table = table.groupby(dimensions).min()
+    if summary_operator == 'std':
+        table = table.groupby(dimensions).std()
+    if summary_operator == 'var':
+        table = table.groupby(dimensions).var()
+    if summary_operator == 'first':
```

Comment:
Just in case if the user wants some examples of the data points that are being compressed. It is not widely asked by users, but it can be added.
ex
top 5 classes based on the marks of the last student in the class.